portfolio/src/main/webapp/comments/CommentRenderer.js

```diff
@@ -0,0 +1,33 @@
+
+class CommentRenderer extends RendererInterface {
+    constructor(targetElement) {
+        super();
+
+        this.target = targetElement;
+    }
+
+    /**
+     * Renders content on the comment section.
+     * @param comments an array with the comment details to be rendered
+     */
+    render(comments) {
+        this.target.innerHTML = "";
+
+        for(const comment of comments) {
+            let commentHTML = `
+                <div class="comment-container">
+                    <div class="comment-header">
+                        <p class="comment-name">${comment.name}</p>
+                        <div class="comment-header-bullet-seperator"></div>
+                        <p class="comment-date">${comment.postDate}</p>
+                    </div>
+                    <div class="comment-body">
+                        <p class="comment-desc">${comment.description}</p>
+                    </div>
+                </div>
+            `;
+
+            this.target.innerHTML += commentHTML;
```

Comment:
Now it is exposed to JavaScript injection.

Reply:
Se comments on the web page ;)

Reply:
Is it better to check for JS injection on the front-end or on the back-end?

Reply:
I made a small research about it and I found out we need to replace < with "&lt;" and > with "&gt;"

Reply:
It depends. In this case I would have done it on the client side.
Another note - js has built in methods to sanitize possibly malicious tags/attributes.